CMakeLists.txt

```diff
@@ -8,6 +8,42 @@
 #   -DEXPERIMENTAL=<ON|OFF>
 #
 
+# Generic function to normalize option variables
+# Usage: normalize_option(VAR_NAME [DEFAULT default_value] [ALLOWED_VALUES val1 val2...])
+function(normalize_option VAR_NAME)
+  set(options "")
+  set(oneValueArgs DEFAULT)
+  set(multiValueArgs ALLOWED_VALUES)
+  cmake_parse_arguments(NORM "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  # Set defaults
+  if(NOT NORM_DEFAULT)
+    set(NORM_DEFAULT "AUTO")
+  endif()
+  if(NOT NORM_ALLOWED_VALUES)
+    set(NORM_ALLOWED_VALUES "ON" "OFF" "AUTO")
+  endif()
+
+  # Normalize truthiness values
+  if(DEFINED ${VAR_NAME})
+    set(var_value ${${VAR_NAME}})
+
+    # Convert various truthiness values to standard ON/OFF
+    if(var_value MATCHES "^(1|[Tt][Rr][Uu][Ee]|[Yy][Ee][Ss])$")
+      set(${VAR_NAME} "ON" PARENT_SCOPE)
+    elseif(var_value MATCHES "^(0|[Ff][Aa][Ll][Ss][Ee]|[Nn][Oo])$")
+      set(${VAR_NAME} "OFF" PARENT_SCOPE)
```

Comment:
Why such complicated logic instead of just giving an error if none of the 3 allowed values are given?

Reply:
So that needs to be fixed *once* and that's it. If it's giving an error, it's easy to spot. Yes, that could be quite a number of places and that's slightly annoying, but I think preferable to making the `cmake` setup much more complex than needed and keeping that forever.

If we need to allow "truthy" values, we should just rely on `cmake` to do the interpretation instead of implementing our own (e.g. like suggested in https://cmake.org/pipermail/cmake/2016-October/064342.html).